### PR TITLE
lint our jsx files

### PR DESCRIPTION
**Pull request by AWare**
This turns on jsxlint, and fixes most of the errors. 

The properties not checked error is reduced to a warning because it will take a lot of work to fix.

### CODE REVIEW

**Comment by AWare** [comment]
image

`react/prop-types` is all 51 warnings. we should consider adding Prop-Types or whether this linter rule should be disabled.

**Comment by paulbrown1982** [comment]
Looks great, just a couple of little bits I'd like to understand more about. Let's chat tomorrow.

File: app/views/fragments/checkout/promoCode.scala.html [hunk 1]
```diff
@@ -19,7 +19,7 @@
         <div><strong>✓&nbsp;&nbsp;Promotion applied</strong></div>
         <div class="js-promo-code-snippet"></div>
         <div class="u-note">
-             <a class="u-link js-promo-code-tsandcs" href="#" target="_blank">See full terms and conditions</a>
+             <a class="u-link js-promo-code-tsandcs" href="#" target="_blank" rel="noopener noreferrer">See full terms and conditions</a>
```
Comment (paulbrown1982) [inline app/views/fragments/checkout/promoCode.scala.html]: What does this mean/do?

File: assets/javascripts/modules/react/customDateRangePicker.jsx [hunk 2]
```diff
@@ -2,79 +2,88 @@ import React from 'react'
...
+
+     displayName = 'CustomDateRangePicker';
+
+     defaultProps = (() => {
         const maxWeeks = 6;
         return {
             remainingDays: (maxWeeks * 7),
             firstSelectableDate: moment(),
-            lastStartDate: moment().add(1, 'year')
+            laststartDate: moment().add(1, 'year')
```
Comment (paulbrown1982) [inline assets/javascripts/modules/react/customDateRangePicker.jsx]: Why was this case changed?

File: app/views/fragments/checkout/promoCode.scala.html (hunk 1 above)
Comment (AWare) [reply app/views/fragments/checkout/promoCode.scala.html]: without it the window object is available to the calling page.https://mathiasbynens.github.io/rel-noopener/

**Comment by AWare** [comment]
This also fixes people being told payment details updated correctly when they did not.

**Comment by paulbrown1982** [comment]
Tested on CODE before merging.

**Pull request merged**